src/config.rs

```diff
@@ -580,6 +601,20 @@ impl Context {
         Ok(())
     }
 
+    /// Set the given config key and make it effective.
+    /// This may restart the IO scheduler. If `None` is passed as a value the value is cleared and
+    /// set to the default if there is one.
+    pub async fn apply_config(&self, key: Config, value: Option<&str>) -> Result<()> {
```

Comment:
Switched to using `set_config_ex()` internally

Reply:
Some calls to `set_config()` need to be changed to `set_config_ex()` at least to avoid Rust complains about recursive async calls. But then it's not clear for the reader why we use both functions internally. To make it the same everywhere and also make it clear that we don't need to restart the IO scheduler, i addded `set_config_internal()` now as per your initial suggestion. This way we don't need to to write `set_config_ex(None, ...)` everywhere.